.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,45 @@
+name: "Docker Image"
+
+on:
+  push:
+    branches:
+      - "main"
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
How will this work? Do I need to connect something in Docker with the GH token?

Reply:
Thanks for all the changes @jpylypiw I just need to understand this one, if you can link me to some documentation, that works.

Reply:
I just read through the documentation and saw this is going into the GitHub Container Registry, so it's not going to Docker Hub, where I thought it was going to, originally.